src/vanilla.ts

```diff
@@ -5,7 +5,11 @@ export type PartialState<T extends State> =
 export type StateSelector<T extends State, U> = (state: T) => U
 export type EqualityChecker<T> = (state: T, newState: unknown) => boolean
 export type StateListener<T> = (state: T) => void
-export type StateSliceListener<T> = (state: T | null, error?: Error) => void
+export type StateSliceListener<T> = (
+  state: T | null,
+  error: Error | null,
+  previousState: T | null
+) => void
```

Comment:
```suggestion
export type StateSliceListener<T> = (slice: T, previousSlice: T) => void
```

If such a breaking change is allowed, I would prefer this.

It doesn't conflict with #37, because we no longer use this in useStore.

Reply:
This looks reasonably safe to me. The original issue was selectors could throw an error because they were called at unexpected times. We didn't want to silence all errors in case it was something they need to handle so we exposed it.

I think it's not used very much.

Reply:
i agree, it's probably more useful to have the previous state.

Reply:
Thanks, guys.
One it's adjusted accordingly, we will release it as a minor update.
Although it's technically a breaking change in the vanilla api, the impact would be tiny.
- it's not documented in readme
- it should not be used very much

Reply:
I changed it to:

```javascript
export type StateSliceListener<T> = (slice: T | null, previousSlice: T | null) => void
```

Because when there is an error we still return `null, null`. Let me know if you want to return something else instead or remove the try-except completely.